### PR TITLE
Allow admins to cancel and reschedule appointments

### PR DESCRIPTION
**Pull request by miriamstriks**
https://github.com/adamzr/mikvah-api/pull/10 for the backend.

https://user-images.githubusercontent.com/10815484/155016150-74636a15-baf3-4908-b435-ad4b0c49c62c.mov

### CODE REVIEW

File: src/app/admin/admin.component.ts [hunk 1]
```diff
@@ -51,9 +76,106 @@ export class AdminComponent implements OnInit {
...
+          } else if(result.event == 'Delete'){
+            this.deleteRowData(result.data);
+          }
+        });
+      });
+  }
+
+  updateRowData(data: {obj: { action: any, id: number, roomType: string}, time: string }){
+    this.appointmentsService.editAppointment(data.obj.id, this.selectedDate + "T" + data.time)
+      .subscribe(response => {
```
Comment (miriamstriks) [inline src/app/admin/admin.component.ts:121]: Should this event be tracked?

Comment (adamzr) [reply src/app/admin/admin.component.ts:121]: You mean with Google Analytics? I guess we could, but as long as this is tracked on the backend, I think that's good.

File: src/app/dialog-box/dialog-box.component.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+     <div class="form-group">
+          <label for="time">Time</label>
+          <select class="form-control" id="time" name="time" [(ngModel)]="local_data.time" #timeField="ngModel" required>
+            <option *ngFor="let time of local_data.times; let i = index" [value]="time.isoTime">{{time.displayTime}}
+            </option>
+          </select>
+        </div>
+  </mat-form-field>
+  <ng-template #elseTemplate>
+    Delete <b>{{local_data.obj.time}}</b> appointment for <b>{{local_data.obj.email}}</b>?
```
Comment (adamzr) [inline src/app/dialog-box/dialog-box.component.html:12]: Will canceling cause the user to be refunded? Either way, please add that to this text: "Deleting the appointment will cause any payments to be refunded." This way the admin knows that this has an effect on the money.

**Review by adamzr — APPROVED** [review]
Looks great! See comment.

File: src/app/dialog-box/dialog-box.component.html (hunk 2 above)
Comment (miriamstriks) [reply src/app/dialog-box/dialog-box.component.html]: Yes, canceling from the admin page follows the same path as canceling from the appointments page.

**Review by adamzr — APPROVED** [review]

**Pull request merged**